django/contrib/auth/forms.py

```diff
@@ -207,19 +217,48 @@ def __init__(self, request=None, *args, **kwargs):
         if self.fields['username'].label is None:
             self.fields['username'].label = capfirst(self.username_field.verbose_name)
 
+    def _request_digest(self, username):
+        try:
+            origin = str(username) + self.request.META.get('REMOTE_ADDR', '')
```

Comment:
So I forgot to mention that checking `REMOTE_ADDR` is not going to be enough. Most serious deployments will have connections proxied so we would need to check for other headers such as `X-Forwarded-For`, etc. The challenge is that this cannot be blanket-enabled as we cannot assume that user's have configured their server correctly and we don't want those headers to be set by the client rather than the proxy. [`django-axes`](https://pypi.org/project/django-axes/) makes use of [`django-ipware`](https://pypi.org/project/django-ipware) for handling identification of client addresses.

We'd probably want have a way of providing a setting pointing to a function that allows users to customise how this cache key is built? Because I doubt that we're going to want to add a (soft) dependency on another external package.

Reply:
Ok. I guess I've always avoided overriding `REMOTE_ADDR` directly, but I can see that other things out there provide the ability or instructions on how to do just that, e.g. [django-ratelimit](https://django-ratelimit.readthedocs.io/en/stable/security.html#middleware), [werkzeug](https://werkzeug.palletsprojects.com/en/1.0.x/middleware/proxy_fix/?highlight=remote_addr#x-forwarded-for-proxy-fix).